service/actions/websocket.js

```diff
@@ -0,0 +1,460 @@
+    }
+
+    dispatch({type: GeneralTypes.WEBSOCKET_SUCCESS}, getState);
+}
+
+function handleClose(connectFailCount, dispatch, getState) {
+    dispatch({type: GeneralTypes.WEBSOCKET_FAILURE, error: connectFailCount}, getState);
+}
+
+function handleEvent(msg, dispatch, getState) {
+    switch (msg.event) {
+    case WebsocketEvents.POSTED:
+    case WebsocketEvents.EPHEMERAL_MESSAGE:
+        handleNewPostEvent(msg, dispatch, getState);
+        break;
+    case WebsocketEvents.POST_EDITED:
+        handlePostEdited(msg, dispatch, getState);
+        break;
+    case WebsocketEvents.POST_DELETED:
+        handlePostDeleted(msg, dispatch, getState);
+        break;
+    case WebsocketEvents.LEAVE_TEAM:
+        handleLeaveTeamEvent(msg, dispatch, getState);
+        break;
+    case WebsocketEvents.USER_ADDED:
+        handleUserAddedEvent(msg, dispatch, getState);
+        break;
+    case WebsocketEvents.USER_REMOVED:
+        handleUserRemovedEvent(msg, dispatch, getState);
+        break;
+    case WebsocketEvents.USER_UPDATED:
+        handleUserUpdatedEvent(msg, dispatch, getState);
+        break;
+    case WebsocketEvents.CHANNEL_VIEWED:
+        handleChannelViewedEvent(msg, dispatch, getState);
+        break;
+    case WebsocketEvents.CHANNEL_DELETED:
+        handleChannelDeletedEvent(msg, dispatch, getState);
+        break;
+    case WebsocketEvents.DIRECT_ADDED:
+        handleDirectAddedEvent(msg, dispatch, getState);
+        break;
+    case WebsocketEvents.PREFERENCE_CHANGED:
+        handlePreferenceChangedEvent(msg, dispatch, getState);
+        break;
+    case WebsocketEvents.STATUS_CHANGED:
+        handleStatusChangedEvent(msg, dispatch, getState);
+        break;
+    }
+}
+
+function handleNewPostEvent(msg, dispatch, getState) {
+    const state = getState();
+    const users = state.entities.users;
+    const channels = state.entities.channels;
+    const teams = state.entities.teams;
+    const {posts} = state.entities.posts;
+    const isActive = state.entities.general.appState;
+    const post = JSON.parse(msg.data.post);
+    const userId = post.user_id;
+    const teamId = msg.data.team_id;
+    const status = users.statuses[userId];
+
+    if (!users.profiles[userId]) {
+        getProfilesHelper([userId], dispatch, getState);
+    }
+
+    if (status !== Constants.ONLINE) {
+        getStatusesHelper([userId], dispatch, getState);
+    }
+
+    if (post.channel_id === channels.currentId) {
+        if (isActive) {
+            updateLastViewedHelper(teamId, post.channel_id, dispatch, getState);
+        } else {
+            getChannelHelper(teamId, post.channel_id, dispatch, getState);
+        }
+    } else if (msg && (teamId === teams.currentId || msg.data.channel_type === Constants.DM_CHANNEL)) {
+        getChannelHelper(teamId, post.channel_id, dispatch, getState);
+    }
+
+    if (post.root_id && !posts[post.root_id]) {
+        Client.getPost(teamId, post.channel_id, post.root_id).then((data) => {
```

Comment:
That was my intention at first but dunno why it wasn't working, I'll try again.. if that's the case I'll use that for everything in here